tests/test.FsWatcher.js

```diff
@@ -0,0 +1,177 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright (c) 2015, Joyent, Inc.
+ */
+
+var execFile = require('child_process').execFile;
+
+var test = require('tape');
+var vmadm = require('vmadm');
+
+var common = require('./common');
+var mocks = require('./mocks');
+var FsWatcher = require('../lib/watchers/fs-watcher');
+
+
+// How frequently to poll the 'events' array when we're waiting for an event.
+var EVENTS_POLL_FREQ = 100; // ms
+
+var events = [];
+var existingVms = [];
+var smartosImageUUID;
+var smartosVmUUID;
+var watcher;
+
+
+function waitEvent(t, evt, vmUuid, eventIdx) {
+    var loops = 0;
+
+    function _waitEvent() {
+        var i;
+
+        if (events.length > eventIdx) {
+            // we've had some new events, check for our create
+            for (i = eventIdx; i < events.length; i++) {
+                if (events[i].vmUuid === vmUuid && events[i].event === evt) {
+                    t.ok(true, 'FsWatcher saw expected ' + evt
```

Comment:
I find in general having test utility functions call `t.ok`, `t.end` and other such functions to be a bit cumbersome, because then for instance you can't really call `waitEvent` twice in the same test, as it would call `t.end` twice.

I would suggest having `waitEvent` call a callback and then the test that called it would call various `t.*` functions.